fix: OPTIC-712: Remove warnings in the app related to injection occur…

luarmr opened the pull request:
Removing warnings in console:

<img width="1863" alt="Screenshot 2024-05-03 at 9 52 27 AM" src="https://github.com/HumanSignal/label-studio/assets/817416/a2191453-2f47-4e8d-8956-c42793a2ef3b">

luarmr committed:
fix: OPTIC-712: Remove warnings in the app related to injection occur before observer

hlomzik reviewed (approved):
This can even be improved further because some of components (`RelationsTab`, `InfoTab`) don't need `store`, so `inject()` can also be removed; plus for the rest `observer()` can be moved to component declaration as well for consistency.
But it's ok like this to fix the bug!

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8989906455
